packages/playground/website/src/main.tsx

```diff
@@ -131,6 +131,53 @@ function Modals() {
 	return null;
 }
 
+/**
+ * Allow the parent window to execute PHP code in Playground.
+ *
+ * ## Example:
+ *
+ * ### Send a message to the Playground iframe:
+ * ```
+ * const frame = document.getElementById('iframe-id');
+ * frame.contentWindow.postMessage({ type: 'php-request', code: `<?php echo 'Hey Playground!';` }, '*');
+ * ```
+ *
+ * ### Receive a message from the Playground iframe:
+ * ```
+ * window.addEventListener('message', (event) => {
+ * 	if(event.data.type === 'php-response') {
+ *		console.log('Message received', event.data.response);
+ * 	}
+ * });
+ *
+ */
+function addPhpRequestListener(playground: PlaygroundClient) {
+	window.addEventListener('message', async (event) => {
+		if (event.data.type === 'php-request') {
+			try {
+				const response = await playground.run({
+					code: event.data.code,
+				});
+				window.parent.postMessage(
+					{
+						type: 'php-response',
+						response: response?.text,
+					},
```

Comment:
It would be useful if the response included the request object, or some other way to match the response back up with the request. Since the request could be large, maybe we could denote a certain key that will be included in the reply?

Reply:
We could do something like this.

```suggestion
				window.parent.postMessage(
					{
						type: 'php-response',
						response: response?.text,
						requestId: request.id 
					},
```

Reply:
That sounds good, we just need to omit this there is no id in the request.